src/PhpSpreadsheet/Worksheet/Table.php

```diff
@@ -0,0 +1,471 @@
+     *
+     * @return string
+     */
+    public function getRange()
+    {
+        return $this->range;
+    }
+
+    /**
+     * Set Table Cell Range.
+     *
+     * @return $this
+     */
+    public function setRange(string $range): self
+    {
+        // extract coordinate
+        [$worksheet, $range] = Worksheet::extractSheetTitle($range, true);
+        if (empty($range)) {
+            //    Discard all column rules
+            $this->columns = [];
+            $this->range = '';
+
+            return $this;
+        }
+
+        if (strpos($range, ':') === false) {
+            throw new PhpSpreadsheetException('Table must be set on a range of cells.');
+        }
+
+        $this->range = $range;
+        //    Discard any column ruless that are no longer valid within this range
+        [$rangeStart, $rangeEnd] = Coordinate::rangeBoundaries($this->range);
+        foreach ($this->columns as $key => $value) {
+            $colIndex = Coordinate::columnIndexFromString($key);
+            if (($rangeStart[0] > $colIndex) || ($rangeEnd[0] < $colIndex)) {
+                unset($this->columns[$key]);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Set Table Cell Range to max row.
+     *
+     * @return $this
+     */
+    public function setRangeToMaxRow(): self
+    {
+        if ($this->workSheet !== null) {
+            $thisrange = $this->range;
+            $range = preg_replace('/\\d+$/', (string) $this->workSheet->getHighestRow(), $thisrange) ?? '';
+            if ($range !== $thisrange) {
+                $this->setRange($range);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get Table's Worksheet.
+     *
+     * @return null|Worksheet
+     */
+    public function getWorksheet()
+    {
+        return $this->workSheet;
+    }
+
+    /**
+     * Set Table's Worksheet.
+     *
+     * @return $this
+     */
+    public function setWorksheet(?Worksheet $worksheet = null)
+    {
+        if ($this->name != '' && $worksheet != null) {
+            $spreadsheet = $worksheet->getParent();
+
+            foreach ($spreadsheet->getWorksheetIterator() as $sheet) {
+                foreach ($sheet->getTableCollection() as $table) {
+                    if ($table->getName() == $this->name) {
+                        throw new PhpSpreadsheetException("Workbook already contains a table named '{$this->name}'");
+                    }
+                }
+            }
+        }
+
+        $this->workSheet = $worksheet;
+
+        return $this;
+    }
+
+    /**
+     * Get all Table Columns.
+     *
+     * @return Table\Column[]
+     */
+    public function getColumns()
+    {
+        return $this->columns;
+    }
+
+    /**
+     * Validate that the specified column is in the Table range.
+     *
+     * @param string $column Column name (e.g. A)
+     *
+     * @return int The column offset within the table range
+     */
+    public function testColumnInRange($column)
```

Comment:
Should this be a public method, accessible from userland code?

Reply:
Refactored 3c4a51acb5b50a2d04817db03d063d34b8a5302f